mlrun/api/api/endpoints/datastore_profile.py

```diff
@@ -60,6 +61,24 @@ async def store_datastore_profile(
             HTTPStatus.BAD_REQUEST.value,
             reason="The project name provided in the URI does not match the one specified in the DatastoreProfile",
         )
+    project_ds_name_private = ds.generate_secret_key(info.name, project_name)
+
+    await mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        mlrun.common.schemas.AuthorizationResourceTypes.secret,
+        project_name,
+        mlrun.common.schemas.SecretProviderName.kubernetes,
+        mlrun.common.schemas.AuthorizationAction.store,
+        auth_info,
+    )
+    await run_in_threadpool(
+        mlrun.api.crud.Secrets().store_project_secrets,
```

Comment:
I'm not crazy about having specific business logic such as storing sensitive info in secrets being handled in the endpoint code. OTOH, at the moment you don't have a specific CRUD object for handling storage profiles, so I guess adding one just for this may be too much of an overhead.
Maybe add a small `TODO` comment here noting that whenever new functionality is added and a CRUD is introduced, this should move there.